src/ResourceManager/RecoveryServices.Backup/RecoveryServicesBackupCmdletBase.cs

```diff
@@ -119,5 +126,36 @@ public override void ExecuteCmdlet()
 
             InitializeAzureBackupCmdlet();
         }
+
+        public AzureRmRecoveryServicesJobBase GetJobObject(string jobId)
+        {
+            return JobConversions.GetPSJob(HydraAdapter.GetJob(jobId));
+        }
+
+        public List<AzureRmRecoveryServicesJobBase> GetJobObject(List<string> jobIds)
+        {
+            List<AzureRmRecoveryServicesJobBase> result = new List<AzureRmRecoveryServicesJobBase>();
+            foreach (string jobId in jobIds)
+            {
+                result.Add(GetJobObject(jobId));
+            }
+            return result;
+        }
+
+        public BackUpOperationStatusResponse WaitForOperation(string url)
+        {
+            // using this directly because it doesn't matter which function we use.
+            // return type is same and currently we are using it in only two places.
+            // protected item and policy.
+            BackUpOperationStatusResponse response = HydraAdapter.GetProtectedItemOperationStatusByURL(url);
```

Comment:
why hardcoded to GetProtectedItemOperationStatusByURL? take this method name as 'Action' input.